Training24/Program.c

```diff
@@ -1,15 +1,36 @@
-// ------------------------------------------------------------------------------------
+// --------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
-// -------------------------------------
-//  Dhaarani Mohanasundaram
-// -------------------------------------
-// Standard Program file in Main
+// Dhaarani Mohanasundaram
 // -------------------------------------
 // Program.c
-// Program on main branch.
-// ------------------------------------------------------------------------------------
-
-const char* GetMessage() {
-return "Hello, World!";
+// Program on A5 branch.
+// Program has functions for sorting an array and searching an element in array.
+// -------------------------------------------------------------------------------
+void BubbleSort (int arr[], int n) {
+   for (int i = 0; i < n - 1; i++) {
+      for (int j = 0; j < n - i - 1; j++) {
+         // Swap if the element found is greater than the next element
+         if (arr[j] > arr[j + 1]) {
+            int temp = arr[j];
+            arr[j] = arr[j + 1];
+            arr[j + 1] = temp;
+         }
+      }
+   }
 }
+
+int BinarySearch (int arr[], int n, int target) {
+   int left = 0, right = n - 1;
+   while (left <= right) {
+      int mid = left + (right - left) / 2;
```

Comment:
This was asked to reason out why mid is left + (right - left) / 2 and not right + left/2. 
The reason why right + left/2 is avoided is that this could cause integer overflow when right and left are large numbers.
[https://stackoverflow.com/questions/6735259/calculating-mid-in-binary-search](url)